appside/model/dbt_worksheet_model.js

```diff
@@ -71,76 +143,77 @@ class DbtWorksheetModelFwd extends Model {
 
     /**
      * Build consent disclosure frames for a DBT worksheet model.
+     * @param consent_questions - list of [question (string), response (bool)]
+     * @return the Frame
      */
-    build_consent_disclosure_frame() {
-        let consent_disclosure = {};
+    build_consent_disclosure_frame(consent_questions) {
+        let consent_frame = {};
 
-        consent_disclosure.title = CONSENT_DISCLOSURE_TITLE;
-        consent_disclosure.template = CONSENT_FRAME_TEMPLATE;
-        consent_disclosure.instructions = CONSENT_DISCLOSURE_INSTRUCTIONS;
+        consent_frame.title = CONSENT_DISCLOSURE_TITLE;
+        consent_frame.response_name = RESPONSE_GENERIC;
+        consent_frame.template = CONSENT_FRAME_TEMPLATE;
+        consent_frame.instructions = CONSENT_DISCLOSURE_INSTRUCTIONS;
 
-        let questions = [];
-        for (let each of CONSENT_DISCLOSURE_QUESTIONS) {
-            questions.push([each, false]);
-        }
-        consent_disclosure.questions = questions;
-        return consent_disclosure;
+        consent_frame.questions = consent_questions;
+        return new ConsentDisclosureFrame(consent_frame, this.logger);
```

Comment:
why are we now storing frames as objects

Reply:
it's so we can log timestamps from each frame on user events such as clicking next.